### PR TITLE
Add repo

### PR DESCRIPTION
**Pull request by GentleDonkey**

### CODE REVIEW

File: internal/admin/repository.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	db *gorm.DB
+}
+
+func NewAdminRepo(db *gorm.DB) *adminRepo {
+	return &adminRepo{
+		db,
+	}
+}
+
+func (a *adminRepo) Login(ta Admin) (*Admin, error, string, int) {
```
Comment (longhuo-0) [inline internal/admin/repository.go:21]: you can query user and compare in service layer

File: internal/shipment/handler.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	r.HandleFunc("/shipments", sh.FindAll).Methods("GET")
+	r.HandleFunc("/shipment/{id}/", sh.FindOne).Methods("GET")
+	r.HandleFunc("/shipment/{id}", sh.Delete).Methods("DELETE")
+	r.HandleFunc("/shipment/{id}", sh.Update).Methods("PATCH")
+}
+
+func (handler *APIShipmentHandler) Create(w http.ResponseWriter, r *http.Request) {
+	tkn := jwt.TokenParse(r)
+	if tkn.Authorization == true {
+		shipment := Shipment{
```
Comment (longhuo-0) [inline internal/shipment/handler.go:33]: we should accept payload as application/json, not form value.
see this link https://www.sohamkamani.com/golang/json/

File: internal/shipment/handler.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		api.NewResponse(w, true, err, message, shipment, code)
+	} else {
+		api.NewResponse(w, false, tkn.Error, tkn.Message, nil, tkn.StatusCode)
+	}
+}
+
+func (handler *APIShipmentHandler) FindAll(w http.ResponseWriter, r *http.Request) {
+	tkn := jwt.TokenParse(r)
+	if tkn.Authorization == true {
+		result, err, message, code := handler.repo.FindAll()
```
Comment (longhuo) [inline internal/shipment/handler.go:51]: err check nedded

File: internal/shipment/repository.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"gorm.io/gorm"
+)
+
+type APIShipmentRepo interface {
+	Create(s Shipment) (error, string, int)
+	FindAll() ([]Shipment, error, string, int)
+	FindOne(id int) (Shipment, error, string, int)
+	Update(s Shipment) (error, string, int)
+	Delete(id int) (error, string, int)
```
Comment (longhuo-0) [inline internal/shipment/repository.go:12]: not a big fun of return http status code in repo layer, code show attach to error

Comment (longhuo-0) [reply internal/shipment/repository.go:12]: please add an index on userId

File: internal/shipment/model.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package shipment
+
+type Shipment struct {
+	ID          string `gorm:"column:ID"          json:"ID"`
+	UserID      string `gorm:"column:UserID"      json:"UserID"`
+	Description string `gorm:"column:Description" json:"Description"` //including purchase date and products, quantity
+	Tracking    string `gorm:"column:Tracking"    json:"Tracking"`
+	Comment     string `gorm:"column:Comment"     json:"Comment"`
+	Date        string `gorm:"column:Date"        json:"Date"` //date of the creation of tracking number
```
Comment (longhuo-0) [inline internal/shipment/model.go:9]: all responses should use id and userId instead of ID and UserID

File: internal/shipment/handler.go (hunk 3 above)
Comment (longhuo) [reply internal/shipment/handler.go:51]: should be decoded use json package

File: internal/shipment/handler.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+	tkn := jwt.TokenParse(r)
+	if tkn.Authorization == true {
+		shipment := Shipment{
+			ID:          r.FormValue("ID"),
+			UserID:      r.FormValue("UserID"),
+			Description: r.FormValue("Description"),
+			Tracking:    r.FormValue("Tracking"),
+			Comment:     r.FormValue("Comment"),
+			Date:        r.FormValue("Date"),
+		}
```
Comment (longhuo-0) [inline internal/shipment/handler.go:40]: please use json package

**Pull request merged**